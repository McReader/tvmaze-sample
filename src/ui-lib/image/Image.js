import PropTypes from "prop-types"

export function Image({
	alt,
	src,
	placeholderWidth = 140,
	placeholderHeight,
	...props
}) {
	let placeholderSrc

	if (!src) {
		const placeholderSize = [placeholderWidth]

		if (placeholderHeight) {
			placeholderSize.push(placeholderHeight)
		}

		placeholderSrc = `https://via.placeholder.com/${placeholderSize.join(
			"x"
		)}?text=No image`
	}

	return <img {...props} alt={alt} src={src || placeholderSrc} />
}

Image.propTypes = {
	alt: PropTypes.string,
	src: PropTypes.string,
	placeholderHeight: PropTypes.number,
	placeholderWidth: PropTypes.number,
}
