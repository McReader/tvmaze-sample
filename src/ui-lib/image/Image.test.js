import renderer from "react-test-renderer";

import { Image } from "./Image";

let tree;
let props;

beforeEach(() => {
    tree = renderer
        .create(<Image {...props} />)
        .toJSON();
});

describe(`when "src" prop is provided`, () => {
    beforeAll(() => {
        props = {
            src: "https://example.com",
        };
    });

    test(`should render image with provided "src"`, () => {
        expect(tree).toMatchSnapshot();
    });
});

describe(`when "src" prop is not provided`, () => {
    beforeAll(() => {
        props = {};
    });

    test(`should render image placeholder with default size`, () => {
        expect(tree).toMatchSnapshot();
    });

    describe(`and "placeholderWidth" is provided"`, () => {
        beforeAll(() => {
            props = {
                placeholderWidth: 200,
            };
        });

        test(`should render square image placeholder with provided size`, () => {
            expect(tree).toMatchSnapshot();
        });
    });

    describe(`and "placeholderWidth" and "placeholderHeight" are provided"`, () => {
        beforeAll(() => {
            props = {
                placeholderWidth: 200,
                placeholderHeight: 140,
            };
        });

        test(`should render image placeholder with provided width and height`, () => {
            expect(tree).toMatchSnapshot();
        });
    });
});
