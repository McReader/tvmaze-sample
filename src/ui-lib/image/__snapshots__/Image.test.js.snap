// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when "src" prop is not provided and "placeholderWidth" and "placeholderHeight" are provided" should render image placeholder with provided width and height 1`] = `
<img
  src="https://via.placeholder.com/200x140?text=No image"
/>
`;

exports[`when "src" prop is not provided and "placeholderWidth" is provided" should render square image placeholder with provided size 1`] = `
<img
  src="https://via.placeholder.com/200?text=No image"
/>
`;

exports[`when "src" prop is not provided should render image placeholder with default size 1`] = `
<img
  src="https://via.placeholder.com/140?text=No image"
/>
`;

exports[`when "src" prop is provided should render image with provided "src" 1`] = `
<img
  src="https://example.com"
/>
`;
